src/main/scala/com/yotpo/metorikku/output/writers/file/HudiOutputWriter.scala

```diff
@@ -321,8 +357,97 @@ class HudiOutputWriter(props: Map[String, Object], hudiOutput: Option[Hudi]) ext
     df
   }
 
+  def manualHiveSync(dataFrame: DataFrame, path: Option[String], manualHiveSyncPartitions: Option[Map[String, String]]): DataFrame = {
+
+    val ss = dataFrame.sparkSession
+    val catalog = ss.catalog
+    val df = getHudiDf(dataFrame, ss, path, manualHiveSyncPartitions)
+
+    val tablesToSync = getTablesToSyncByStorageType(hudiOutput, hudiOutputProperties.tableName.get)
+    val tableType = CatalogTableType("EXTERNAL")
+    for ((table, tableInputFormat) <- tablesToSync) {
+      val identifier = new TableIdentifier(table, Option(catalog.currentDatabase))
+      val storage = new CatalogStorageFormat(Option(new URI(path.get)),
+        Option(tableInputFormat),
+        Option(classOf[MapredParquetOutputFormat].getName),
+        Option(classOf[ParquetHiveSerDe].getName),
+        false,
+        Map[String, String]())
+
+      val tableDefinition = new CatalogTable(identifier, tableType, storage, df.schema,
+        partitionColumnNames = manualHiveSyncPartitions match {
+          case Some(partitions) => Seq(partitions.keySet.toSeq: _*)
+          case _ => Seq.empty
+        })
+
+      val schema =  manualHiveSyncPartitions match {
+        case Some(manualHiveSyncPartitions) => {
+          ss.sharedState.externalCatalog.createTable(tableDefinition, true)
+          df.drop(manualHiveSyncPartitions.keySet.toList: _*).schema
+        }
+        case _ => {
+          ss.sharedState.externalCatalog.dropTable(catalog.currentDatabase, table, true, true)
+          ss.sharedState.externalCatalog.createTable(tableDefinition, false)
+          df.schema
+        }
+      }
+      ss.sharedState.externalCatalog.alterTableDataSchema(catalog.currentDatabase, table, schema)
+      ss.sharedState.externalCatalog.alterTable(tableDefinition)
+
+
+      // Handle partitions
+      manualHiveSyncPartitions match {
+        case Some(partitions) => {
+          partitions.foreach( partition => {
+            val partitionStorage = new CatalogStorageFormat(Option(new URI(path.get + "/" + partition._2)),
+              Option(tableInputFormat),
+              Option(classOf[MapredParquetOutputFormat].getName),
+              Option(classOf[ParquetHiveSerDe].getName),
+              false,
+              Map[String, String]())
+            ss.sharedState.externalCatalog.createPartitions(ss.catalog.currentDatabase, table,
+              Seq(CatalogTablePartition(Map(partition._1 -> partition._2), partitionStorage)), true)
+          }
+          )
+
+        }
+        case _ => {
+          ss.sharedState.externalCatalog.listPartitionNames(ss.catalog.currentDatabase, table).size match {
+            case 0 => None
+            case _ => ss.sharedState.externalCatalog.listPartitions(ss.catalog.currentDatabase, table).foreach( part => {
```

Comment:
do we still need this at all? if you are dropping the table in case we don't have a partition definition?